timelock-agent/src/main/java/com/palantir/timelock/config/PaxosInstallConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.timelock.config;
+
+import java.io.File;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+
+@JsonDeserialize(as = ImmutablePaxosInstallConfiguration.class)
+@JsonSerialize(as = ImmutablePaxosInstallConfiguration.class)
+@Value.Immutable
+public interface PaxosInstallConfiguration extends TimeLockAlgorithmInstallConfiguration {
+    @JsonProperty("data-directory")
+    @Value.Default
+    default File dataDirectory() {
+        // TODO: should this value be something like "mnt/timelock/paxos" given we'll be
+        // given a persisted volume that will be mounted inside $SERVICE_HOME in k8s/docker
+        // TODO: should we just have a generic "dataDirectory" field at root TimeLockInstallConfiguration
+        // level and delete this entire file?
+        return new File("var/data/paxos");
+    }
```

Comment:
Happy to change the default to be mnt/... for k8s.

I'm open to the data directory field. Subsequently if we need to restore some kind of install configuration for algorithms it should be an additive change anyways